### PR TITLE
Migrate image pull secrets associated with default service account

### PR DESCRIPTION
**Pull request by ram-infrac**
**What type of PR is this?**
>bug

**What this PR does / why we need it**:
We don't maintain link between secrets while migrating `default` service account. This PR check and update imagepullSecrets for default service account

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:

**Does this change need to be cherry-picked to a release branch?**:
yes

### CODE REVIEW

File: pkg/migration/controllers/migration.go [hunk 1]
```diff
@@ -676,6 +676,66 @@ func (m *MigrationController) updateResourceStatus(
...
+		if err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Delete(sourceSA.GetName(), &metav1.DeleteOptions{}); err != nil {
+			return err
+		}
+		if _, err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Create(&sourceSA); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	log.MigrationLog(migration).Infof("Updating default service account with image pull secrets")
```
Comment (disrani-px) [inline pkg/migration/controllers/migration.go]: Print the namespace of the service account too

File: pkg/migration/controllers/migration.go [hunk 2]
```diff
@@ -676,6 +676,66 @@ func (m *MigrationController) updateResourceStatus(
...
+	// merge service account resource for default namespaces
+	destSA, err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Get(sourceSA.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	destSA.ImagePullSecrets = append(destSA.ImagePullSecrets, sourceSA.ImagePullSecrets...)
```
Comment (disrani-px) [inline pkg/migration/controllers/migration.go]: Can you confirm that this works fine even if the destination service account already has imagePullSecrets with the same name.

Comment (ram-infrac) [reply pkg/migration/controllers/migration.go]: it works even though destination service account already has imagePullSecrets, it just add duplicate entry under image pull secrets 
eg. 
```
Name:                default
Namespace:           default
Labels:              <none>
Annotations:         <none>
Image pull secrets:  test-dr
                     test-dr
Mountable secrets:   default-token-44krf
Tokens:              default-token-44krf
Events:              <none>
```

So I added commit to not duplicated already present `imagepullsecrets`

File: pkg/migration/controllers/migration.go [hunk 3]
```diff
@@ -676,6 +676,77 @@ func (m *MigrationController) updateResourceStatus(
...
+	}
+
+	log.MigrationLog(migration).Infof("Updating default service account(namespace : %v) with image pull secrets", sourceSA.GetNamespace())
+	// merge service account resource for default namespaces
+	destSA, err := adminClient.CoreV1().ServiceAccounts(sourceSA.GetNamespace()).Get(sourceSA.GetName(), metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	found := false
```
Comment (disrani-px) [inline pkg/migration/controllers/migration.go]: `found` needs to be initialized to `false` inside the first loop below

Comment (ram-infrac) [reply pkg/migration/controllers/migration.go]: yeah, missed it

**Pull request merged**